src/signal-tracker.ts

```
@@ -0,0 +1,553 @@
+                    const elapsed = Date.now() - this.pendingProbeStartTime;
+                    logger.debug(`Probe timeout after ${elapsed}ms`);
+                    this.markDeviceOffline(this.targetNumber, elapsed);
+                    this.pendingProbeStartTime = null;
+                    this.pendingProbeTimeout = null;
+                    if (this.probeResolve) {
+                        this.probeResolve();
+                        this.probeResolve = null;
+                    }
+                }
+            }, 15000);
+        });
```

Biome (2.1.2)

[error] Promise executor functions should not be `async`.



(lint/suspicious/noAsyncPromiseExecutor)